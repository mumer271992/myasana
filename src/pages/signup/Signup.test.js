import React from 'react';
import { shallow } from 'enzyme';
import Signup from './Signup';


describe('Signup page component test suit', () => {
  let wrapper;
  const signUp = jest.fn();

  beforeEach(() => {
    wrapper = shallow(<Signup signupUser={signUp} />);
  });

  it('Signup page rendered successfully', () => {
    expect(wrapper.length).toEqual(1);
    expect(wrapper).toMatchSnapshot();
  });

  it('Clicking on signin button should call signin function passed in props from Login High Order function', () => {
    const spy = jest.spyOn(wrapper.instance(), 'onSubmit').mockImplementation(jest.fn());
    wrapper.instance().forceUpdate();
    wrapper.find('[data-test="signup-form"]').simulate('submit');
    expect(spy).toHaveBeenCalled();
  });
});
