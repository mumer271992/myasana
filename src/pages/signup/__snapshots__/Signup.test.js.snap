// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Signup page component test suit Signup page rendered successfully 1`] = `
<div
  className="page login-page"
>
  <h2
    className="text-center logo"
  >
    My Asana
  </h2>
  <div
    className="form-container"
  >
    <h2
      className="text-center"
    >
      Sign Up
    </h2>
    <form
      data-test="signup-form"
      onSubmit={[Function]}
    >
      <div
        className="form-group"
      >
        <label>
          Email address
        </label>
        <input
          className="form-control"
          id="email"
          name="email"
          onChange={[Function]}
          placeholder="Enter email"
          type="email"
          value=""
        />
        <small
          className="form-text text-muted"
          id="emailHelp"
        >
          We'll never share your email with anyone else.
        </small>
      </div>
      <div
        className="form-group"
      >
        <label>
          Password
        </label>
        <input
          className="form-control"
          id="password"
          name="password"
          onChange={[Function]}
          placeholder="Password"
          type="password"
          value=""
        />
      </div>
      <button
        className="btn btn-primary"
        type="submit"
      >
        Sign Up
      </button>
      <p>
        Already have an account?
        <Link
          replace={false}
          to="/signin"
        >
          Sign In
        </Link>
      </p>
    </form>
  </div>
</div>
`;
